haystack/nodes/prompt/prompt_node.py

```diff
@@ -0,0 +1,822 @@
+
+        kwargs_with_defaults = self.model_input_kwargs
+        if kwargs:
+            kwargs_with_defaults.update(kwargs)
+        payload = {
+            "model": self.model_name_or_path,
+            "prompt": prompt,
+            "suffix": kwargs_with_defaults.get("suffix", None),
+            "max_tokens": kwargs_with_defaults.get("max_tokens", self.max_length),
+            "temperature": kwargs_with_defaults.get("temperature", 0.7),
+            "top_p": kwargs_with_defaults.get("top_p", 1),
+            "n": kwargs_with_defaults.get("n", 1),
+            "stream": False,  # no support for streaming
+            "logprobs": kwargs_with_defaults.get("logprobs", None),
+            "echo": kwargs_with_defaults.get("echo", False),
+            "stop": kwargs_with_defaults.get("stop", None),
+            "presence_penalty": kwargs_with_defaults.get("presence_penalty", 0),
+            "frequency_penalty": kwargs_with_defaults.get("frequency_penalty", 0),
+            "best_of": kwargs.get("best_of", 1),
+            "logit_bias": kwargs.get("logit_bias", {}),
+        }
+        headers = {"Authorization": f"Bearer {self.api_key}", "Content-Type": "application/json"}
+        response = requests.request("POST", self.url, headers=headers, data=json.dumps(payload), timeout=30)
+        res = json.loads(response.text)
+
+        if response.status_code != 200:
+            openai_error: OpenAIError
+            if response.status_code == 429:
+                openai_error = OpenAIRateLimitError(f"API rate limit exceeded: {response.text}")
+            else:
+                openai_error = OpenAIError(
+                    f"OpenAI returned an error.\n"
+                    f"Status code: {response.status_code}\n"
+                    f"Response body: {response.text}",
+                    status_code=response.status_code,
+                )
+            raise openai_error
+
+        responses = [ans["text"].strip() for ans in res["choices"]]
+        return responses
+
+    @classmethod
+    def supports(cls, model_name_or_path: str) -> bool:
+        return any(m for m in ["ada", "babbage", "davinci", "curie"] if m in model_name_or_path)
+
+
+class PromptModel(BaseComponent):
+    """
+    The PromptModel class is a component that uses a pre-trained model to generate text based on a prompt. Out of
+    the box, it supports two model invocation layers: Hugging Face transformers and OpenAI, with the ability to
+    register additional custom invocation layers.
+
+    Although it is possible to use PromptModel to make prompt invocations on the underlying model, please use
+    PromptNode for interactions with the model. PromptModel instances are the practical approach for multiple
+    PromptNode instances to use a single PromptNode and thus save computational resources.
+    """
+
+    outgoing_edges = 1
+
+    def __init__(
+        self,
+        model_name_or_path: str = "google/flan-t5-base",
+        max_length: Optional[int] = 100,
```

Comment:
If the max_length value is optional I think the default value should be None. What do you think?

If we need the default value to be set then I would recommend removing the Optional type.